src/DGtal/geometry/volumes/BoundedLatticePolytope.ih

```diff
@@ -0,0 +1,765 @@
+}
+
+//-----------------------------------------------------------------------------
+template <typename TSpace>
+typename DGtal::BoundedLatticePolytope<TSpace>::Self&
+DGtal::BoundedLatticePolytope<TSpace>::
+operator+=( UnitCell c )
+{
+  for ( Dimension i = 0; i < c.dims.size(); ++i )
+    *this += UnitSegment( c.dims[ i ] );
+  return *this;
+}
+
+//-----------------------------------------------------------------------------
+template <typename TSpace>
+typename DGtal::BoundedLatticePolytope<TSpace>::Self&
+DGtal::BoundedLatticePolytope<TSpace>::
+operator+=( RightStrictUnitCell c )
+{
+  for ( Dimension i = 0; i < c.dims.size(); ++i )
+    *this += RightStrictUnitSegment( c.dims[ i ] );
+  return *this;
+}
+
+//-----------------------------------------------------------------------------
+template <typename TSpace>
+typename DGtal::BoundedLatticePolytope<TSpace>::Self&
+DGtal::BoundedLatticePolytope<TSpace>::
+operator+=( LeftStrictUnitCell c )
+{
+  for ( Dimension i = 0; i < c.dims.size(); ++i )
+    *this += LeftStrictUnitSegment( c.dims[ i ] );
+  return *this;
+}
+
+//-----------------------------------------------------------------------------
+template <typename TSpace>
+typename DGtal::BoundedLatticePolytope<TSpace>::Integer
+DGtal::BoundedLatticePolytope<TSpace>::
+count() const
+{
+  Integer nb = 0;
+  for ( Point p : D )
+    nb += isDomainPointInside( p ) ? NumberTraits<Integer>::ONE : NumberTraits<Integer>::ZERO;
+  return nb;
+}
+
+//-----------------------------------------------------------------------------
+template <typename TSpace>
+typename DGtal::BoundedLatticePolytope<TSpace>::Integer
+DGtal::BoundedLatticePolytope<TSpace>::
+countInterior() const
+{
+  Integer nb = 0;
+  for ( Point p : D )
+    nb += isInterior( p ) ? NumberTraits<Integer>::ONE : NumberTraits<Integer>::ZERO;
+  return nb;
+}
+//-----------------------------------------------------------------------------
+template <typename TSpace>
+typename DGtal::BoundedLatticePolytope<TSpace>::Integer
+DGtal::BoundedLatticePolytope<TSpace>::
+countBoundary() const
+{
+  Integer nb = 0;
+  for ( Point p : D )
+    nb += isBoundary( p ) ? NumberTraits<Integer>::ONE : NumberTraits<Integer>::ZERO;
+  return nb;
+}
+//-----------------------------------------------------------------------------
+template <typename TSpace>
+typename DGtal::BoundedLatticePolytope<TSpace>::Integer
+DGtal::BoundedLatticePolytope<TSpace>::
+countWithin( Point lo, Point hi ) const
+{
+  Integer nb = 0;
+  Domain D1( lo.sup( D.lowerBound() ), hi.inf( D.upperBound() ) );
+  for ( Point p : D1 )
+    nb += isDomainPointInside( p ) ? NumberTraits<Integer>::ONE : NumberTraits<Integer>::ZERO;
+  return nb;
+}
+//-----------------------------------------------------------------------------
+template <typename TSpace>
+typename DGtal::BoundedLatticePolytope<TSpace>::Integer
+DGtal::BoundedLatticePolytope<TSpace>::
+countUpTo( Integer max) const
+{
+  Integer nb = 0;
+  for ( Point p : D ) {
```

Comment:
```suggestion
  for ( Point &p : D ) {
```

Reply:
I cannot do this (because of the way iterators are written in HyperRectDomain, but I have written "const Point& p : D" everywhere.